linera-core/src/client/chain_state.rs

```diff
@@ -161,7 +161,7 @@ impl ChainState {
         self.pending_blobs.clear();
     }
 
-    pub fn preparing_block(&self) -> Arc<Mutex<()>> {
-        self.preparing_block.clone()
+    pub fn client_mutex(&self) -> Arc<Mutex<()>> {
```

Comment:
Why is this `pub` ?

Reply:
Because the `ChainClient` has to use it. I can make it `pub(super)`.

Reply:
This whole struct is private to `linera_core::client`.  I'm usually suspicious of `pub(super)`, `pub(crate)` and friends because, even though they're sometimes necessary (especially for use in macros), they imply non-local knowledge of the structure the code is embedded into.  The upshot is that types look different (expose different behaviour) depending on where they're imported, which usually just results in a lot of spurious changes when refactoring, but coupled with conditional compilation could lead to some hard-to-spot compilation breakages.

Reply:
Yes, I also prefer just using `pub` in these cases. Happy to revert this in a later PR.

Reply:
As a rule, public APIs need to be minimized.

Reply:
I'd argue that they are, since `ChainState` itself is only visible in the `client` module, and not exported further. So _effectively_ all it's methods are `pub(super)` anyway.